Lsコマンドを作ろうの課題提出

YudaiNoda3576 opened the pull request:
## 目的
lsコマンドを作ろうの課題のレビューを頂くこと

## 達成条件
レビュワーからマージ許可を頂くこと

## このプルリクでできるようになること
下記の仕様に沿った形で引数を取らないlsコマンドと同様の挙動が再現できる

## 仕様
* gemを使わずにRubyの標準ライブラリのみで実装すること
* 二つ以上のメソッドを自分で定義すること（メソッドの作り方を学ぶため）
* 横に最大3列を維持して表示する
  * ただし、「3列から5列に仕様変更してください」または「3列から100列に変えてください」とあとから言われても必要最小限の変更で対応できるようなロジックにしておくこと

## このプルリクでしないこと
* 引数にファイルやディレクトリを指定可能にする。
* 半角英数字以外のファイル名（ひらがなや漢字など）を持つファイルがあっても表示が崩れない。
* mac拡張属性（@マークで表示される）に対応する。

## レビュー希望箇所
* メソッド・変数の命名について違和感のある箇所があるかどうかご確認頂きたいです
* Rdocの書き方について違和感のある箇所があるかどうかご確認頂きたいです
* コメントがなくても意図が伝わるコードになっているかご確認頂きたいです
## 懸念事項
子階層のファイル・ディレクトリを取得→３要素ずつの２次元配列に詰める→二次元配列をソートして配列に詰める→３要素ずつの２次元配列に詰める→出力する

このような実装フローになっていますが、無駄が多いと感じています。理想は、子階層のファイル・ディレクトリを取得→３要素ずつソートして配列に詰める→出力するとしたかったのですが、出力順に並び替えるソートの方法が思いつきませんでした。この点について実装例などございましたらご教示頂きたいです 🙏🏻

YudaiNoda3576 committed:
feat

引数を取らないlsコマンドを実装

YudaiNoda3576 committed:
refactor

各処理をメソッドに切り出す修正

yuuu commented on 05.ls/ls.rb:
```diff
@@ -0,0 +1,66 @@
+  # @return [Array]
+  def output_arry_factory(arry)
+    results = []
+    arry.each_slice(MAX_COLUMN) do |items|
```
ここを `arry.each_slice(MAX_COLUMN).map` とすると `results` という変数が不要となるのでよりスッキリしますね。

こちらも同様です。
https://github.com/YudaiNoda3576/ruby-practices/pull/5/files#diff-8d377e81683a20ae5ea549de93212074e856622b2543e434269d42607f38701bR34

yuuu commented on 05.ls/ls.rb:
```diff
@@ -0,0 +1,66 @@
+  def sorted_arry_factory(row, sliced_by_row)
+    tmp_arry = []
+    i = 0
+    row.times do
```
ここはブロックで0オリジンの数値を取ることができるので、 `i` を自分で宣言する必要はないですね。

```ruby
row.times do |i|
  # iには0, 1, 2...が入る
end
```

もしくはここも範囲リテラルを使うことで `map` とすることができます。

```ruby
(0...row).map do |i|
  sliced_by_row.map do |item|
    item[i]
  end
end
```

yuuu commented on 05.ls/ls.rb:
```diff
@@ -0,0 +1,66 @@
+
+    child_max_length = children.map(&:length).max
+
+    output_row = (children.size % 3).positive? ? children.size / 3 + 1 : children.size / 3
```
[Float#ceil](https://docs.ruby-lang.org/ja/latest/method/Float/i/ceil.html) を使うとここでやりたいことをもっと簡潔に書けるかもしれません 👍

YudaiNoda3576 committed:
Merge branch 'main' of https://github.com/YudaiNoda3576/ruby-practices into ls

YudaiNoda3576 committed:
fix：レビュー箇所の修正

・output_arry_factoryメソッドに定義していたresults変数が不要だったため、削除した
・sorted_arry_factoryに定義されていたi変数が不要だったため、削除した
・output_row変数の算出方法を修正した
